worker/src/runner/container_rt.rs

```diff
@@ -31,23 +35,45 @@ impl ContainerRuntime {
             {
                 let error = e.to_string();
                 handle
-                    .report_instance_status(Status::FailedToStart { error })
+                    .report_instance_status(spec.instance_id, Status::FailedToStart { error })
                     .await;
                 return;
             }
 
-            // healthcheck verifies if service is running on established `PORT`
-            handle.report_instance_status(Status::Started).await;
+            // Healthcheck verifies if service is running on established `PORT`
+            handle
+                .report_instance_status(spec.instance_id, Status::Started)
+                .await;
 
             if let Err(e) = this.wait_container(&container_name).await {
                 let error = e.to_string();
                 handle
-                    .report_instance_status(Status::Crashed { error })
+                    .report_instance_status(spec.instance_id, Status::Crashed { error })
                     .await;
                 return;
             }
 
-            handle.report_instance_status(Status::Terminated).await;
+            handle
+                .report_instance_status(spec.instance_id, Status::Terminated)
+                .await;
+        });
+    }
+
+    pub fn terminate_instance(&self, id: InstanceId) {
+        let this = self.clone();
+        tokio::spawn(async move {
+            let name = format!("instance-{id:?}");
+            loop {
+                match this.stop_container(&name).await {
+                    Ok(()) => {
+                        info!("Successfully stopped container!");
+                        break;
+                    }
+                    Err(e) => {
+                        info!("Failed to stop container: {:?}. Retrying...", e);
+                    }
```

Comment:
Yeah, it would be on this `loop` forever, I'll add a validation on `Runner` before even entering the `container_runtime`, to avoid lost tasks.

Also, I am thinking on adding an `attempt_limit` to the loop, maybe 5, wdyt?

Reply:
We are probably going to need something a little more involved when we implemented termination (also, in another PR).

Reply:
Also, take a look at exponential backoff retry strategy.